realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -128,9 +126,82 @@ void RealSenseNodeFactory::onInit()
                     {
                     if (info.was_removed(_device))
                     {
-                    ROS_FATAL("The device has been disconnected! Terminating RealSense Node...");
-                    ros::shutdown();
-                    exit(1);
+                    ROS_WARN("The device has been disconnected! Looking for a new RealSense...");
+                    rs2::device retDev;
+                    std::string empty("");
+                    std::string serial_no("");
+
+                    try
+                    {
+                      serial_no = retDev.get_info(RS2_CAMERA_INFO_SERIAL_NUMBER);
+                    }
+                    catch(const std::exception& ex)
+                    {
+                      serial_no = "";
+                    }
+
+                    while(serial_no == empty)
+                    {
+                      ROS_WARN("Looking for new realsense");
+                      retDev = getDevice(serial_no);
+                    }
+                    auto nh = getNodeHandle();
+                    auto privateNh = getPrivateNodeHandle();
+                    _device = retDev;
+                    auto pid_str = _device.get_info(RS2_CAMERA_INFO_PRODUCT_ID);
```

Comment:
IINM, lines from here are basically mere copy of the portion from [L209](https://github.com/plusone-robotics/realsense/blob/30d2fadefecc9fb5b9a1e40e3b7820400691636d/realsense2_camera/src/realsense_node_factory.cpp#L209), so this duplicates the maintenance. Why not extract the logic and create a function?

Reply:
Good catch. Fixed.